File: frontend-templateshop/src/services/templateService.js
import api from './api';

// Hàm gọi API để lấy tất cả các template
export const getAllTemplates = () => {
  return api.get('/templates');
};

export const getTemplateBySlug = (slug) => {
  return api.get(`/templates/${slug}`);
};

// --- API cho Admin ---
export const createTemplate = (formData) => {
  // Thêm /admin vào đường dẫn
  return api.post('/admin/templates', formData, {
    headers: {
      'Content-Type': 'multipart/form-data',
    },
  });
};

export const updateTemplate = (id, formData) => {
  // Thêm /admin vào đường dẫn
  return api.put(`/admin/templates/${id}`, formData, {
    headers: {
      'Content-Type': 'multipart/form-data',
    },
  });
};

export const deleteTemplate = (id) => {
  // Thêm /admin vào đường dẫn
  return api.delete(`/admin/templates/${id}`);
};

File: backend-templateshop/src/main/java/com/nhanit/backend_templateshop/service/TemplateService.java
package com.nhanit.backend_templateshop.service;

import java.util.List;

import org.springframework.web.multipart.MultipartFile;

import com.nhanit.backend_templateshop.dto.request.CreateTemplateRequest;
import com.nhanit.backend_templateshop.dto.request.UpdateTemplateRequest;
import com.nhanit.backend_templateshop.dto.response.TemplateResponse;

public interface TemplateService {

  // Phương thức để tạo một template mới, bao gồm cả việc upload file
  TemplateResponse createTemplate(CreateTemplateRequest request, MultipartFile file, MultipartFile thumbnail);

  List<TemplateResponse> getAllTemplates();

  TemplateResponse getTemplateById(long templateId);

  TemplateResponse updateTemplate(Long templateId, UpdateTemplateRequest request, MultipartFile file,
      MultipartFile thumbnail);

  void deleteTemplate(Long templateId);

  TemplateResponse getTemplateBySlug(String slug);

}
